NorthstarDLL/scripts/client/scriptserverbrowser.cpp

```diff
@@ -6,299 +6,105 @@
 
 // functions for viewing server browser
 
-ADD_SQFUNC("bool", NSIsMasterServerAuthenticated, "", "", ScriptContext::UI)
-{
-	g_pSquirrel<context>->pushbool(sqvm, g_pMasterServerManager->m_bOriginAuthWithMasterServerDone);
-	return SQRESULT_NOTNULL;
-}
-
-ADD_SQFUNC("void", NSRequestServerList, "", "", ScriptContext::UI)
-{
-	g_pMasterServerManager->RequestServerList();
-	return SQRESULT_NULL;
-}
-
-ADD_SQFUNC("bool", NSIsRequestingServerList, "", "", ScriptContext::UI)
-{
-	g_pSquirrel<context>->pushbool(sqvm, g_pMasterServerManager->m_bScriptRequestingServerList);
-	return SQRESULT_NOTNULL;
-}
-
-ADD_SQFUNC("bool", NSMasterServerConnectionSuccessful, "", "", ScriptContext::UI)
-{
-	g_pSquirrel<context>->pushbool(sqvm, g_pMasterServerManager->m_bSuccessfullyConnected);
-	return SQRESULT_NOTNULL;
-}
-
-ADD_SQFUNC("int", NSGetServerCount, "", "", ScriptContext::UI)
+ADD_SQFUNC("array<ServerInfo>", NSGetGameServers, "", "", ScriptContext::UI)
 {
-	g_pSquirrel<context>->pushinteger(sqvm, g_pMasterServerManager->m_vRemoteServers.size());
-	return SQRESULT_NOTNULL;
-}
-
-ADD_SQFUNC("string", NSGetServerName, "int serverIndex", "", ScriptContext::UI)
-{
-	SQInteger serverIndex = g_pSquirrel<context>->getinteger(sqvm, 1);
-
-	if (serverIndex >= g_pMasterServerManager->m_vRemoteServers.size())
+	g_pSquirrel<context>->newarray(sqvm, 0);
+	for (size_t i = 0; i < g_pMasterServerManager->m_vRemoteServers.size(); i++)
 	{
-		g_pSquirrel<context>->raiseerror(
-			sqvm,
-			fmt::format(
-				"Tried to get name of server index {} when only {} servers are available",
-				serverIndex,
-				g_pMasterServerManager->m_vRemoteServers.size())
-				.c_str());
-		return SQRESULT_ERROR;
-	}
+		RemoteServerInfo remoteServer = g_pMasterServerManager->m_vRemoteServers[i];
```

Comment:
This is not really needed but if you make this a const ref the code will be optimized better